src/vt_flash/vt_flash.jl

```diff
@@ -326,7 +328,7 @@ function __vt_flash_init_conc_choose(
     Dmin = T(Inf)
     index_min = -1
     for (i, state) in enumerate(vt_stab_tries)
-        if !state.isstable && state.energy_density < Dmin
+        if state.converged && !state.isstable && state.energy_density < Dmin
```

Comment:
Choosing the result with least energy density now explicitly checks if stability result is source of truth (converged flag).